### ISSUE TITLE
Centralised Occupancy of Beds in Private and Public hospitals

### ISSUE THREAD

**Issue by jigar1101** [nic-delhi/AarogyaSetu_Android#330]
In these times, there have been a lot of cases where we have seen that people are unable to find beds.
They have to call each and every public or private hospital in their area to check the availability of bed to find that there are no beds available.

We can have an addon feature in the app, where the users looking to get admitted can check for the availability of the beds.

I know this is a big ask as we don't have a centralized system in place which connects all the public and private hospitals but at least we can take a step towards it.

The setup we can start with.

- On the hospital side (Updating the bed availability, real-time or near real-time)

  - Option 1 - Updating the availability of the bed by the hospital representative every 1/2 hour. Since each hospital has software which they use for billing (they are no new to computers), we can provide them with a link where they can update the bed count in the respective hospital. (For example, the time frame of 1 hour means, the hospitals have to update it only 24 times in a day and we can build such a good UI/UX that they can do it within 1 minute each time (to save their time), thus making a time loss of fewer than 30 mins - which is worth it)

  - Option 2 - If hospitals cannot update it for some reason, we can crowdsource this, a model similar to [covid19india-react](https://github.com/covid19india/covid19india-react)
This will help us gather all the data we need.

The first option, we will be able to execute only if a centralized government body like NIC enforces the hospitals to update the details of the bed every hour or the hospitals are willing to do it (maybe we can reach out to them)

- In the App or on the dashboard [here](https://www.covid19india.org/)

  - We can show the availability of the hospitals based on the user's location, along with search and filter capabilities.

If you'd like to contribute let's start brainstorming.
It would be very helpful if developers, UI/UX guys can give their views on how we can build this system and interpret this data in the app.

This can be the start of a centralized infrastructure in the healthcare sector of our country, expanding to the user's complete health history.

Kindly upvote the issue with 👍 if you think there's a need for it and help brainstorm this issue before it gets started.

**Comment by gunjdesai** [comment]
+1

**Comment by NBallaney** [comment]
+1

**Comment by jaypoojara** [comment]
+1

**Comment by raghavendrahacker** [comment]
@jigar1101 Nice! It can be a good initiative...

**Comment by digikar99** [comment]
Is anyone known to be working on this in any of the 1.4k forks?

**Comment by jigar1101** [comment]
@digikar99 Not sure but when I searched on GitHub I was unable to find anything about it.

**Comment by digikar99** [comment]
Jeez... https://github.com/nic-delhi/AarogyaSetu_Android/issues/440

**Comment by digikar99** [comment]
Is "Covid-19 facilities in States & Union Territories" link at https://www.mohfw.gov.in sufficient?

**Comment by jigar1101** [comment]
Does this contain the Availability of Beds in Private and Public hospitals? I couldn't find it.

**Comment by digikar99** [comment]
Under the appropriate state under [Covid-19 facilities in States & Union Territories](Covid-19 facilities in States & Union Territories) (I assume you did do a Ctrl+F on the MoHFW website), say [Maharashtra](https://arogya.maharashtra.gov.in/1177/Dedicated-COVID-Facilities-Status), the [Facility-Wise](https://arogya.maharashtra.gov.in/pdf/Cat%20I%20Facility%2028%20June%202020.pdf) list lists the total beds. Unfortunately, yes, the number of beds actually free is not available; I think I jumped things on my part.

**Comment by digikar99** [comment]
However, some individual cities / states, seem to be doing their part - [Delhi](https://delhifightscorona.in/data/hospital-beds/) for example; but it seems that [this information is insufficient to avoid chaos in the absence of accountability](https://youtu.be/6lrPggGiGaw) - even though it, indeed, is better than nothing.

**Comment by jigar1101** [comment]
Yes, this is good information but when someone needs to visit the hospitals, they would like to know the current availability which helps them go straight to the available hospital.

Yes, the Delhi government had recently launched an app and a website for showing the real-time availability, it is something that we need but it has to be across the country and not just restricted to Delhi.

That's where NIC-Delhi can play an important role in bringing everyone on one platform.

**Comment by digikar99** [comment]
Yes, at this point, I'm not discounting the need for this service country-wide. 

By nic-delhi, are you referring to this github organization - cause, this repository looks abandoned due to either technical or non-technical reasons - or I didn't get you; could you elaborate?

**Comment by jigar1101** [comment]
Yes, I meant this Github organization.
I'm not aware of any other platform that can have access to hospitals across the country.
If you are aware of any other platform or organization, please tag them, maybe then they could help reach out to the hospitals.

**Comment by digikar99** [comment]
Unfortunately, no, I do not know of any such organizations.

**Comment by jigar1101** [comment]
Cool, no worries, appreciate your effort in researching on the issue.
